### PR TITLE
4044 Properly catch custom ES syntax errors in the V4 Search API

### PR DESCRIPTION
**Pull request by albertisfu**
- As described in #4044, this PR ensures that the **`UnbalancedParenthesesQuery`**, **`UnbalancedQuotesQuery`**, and **`BadProximityQuery`** exceptions are properly caught in the V4 Search API, and the corresponding **`APIException`** error is raised.

  The error **`detail`** is the same as for the original exception:

  - **UnbalancedParenthesesQuery**: The query contains unbalanced parentheses.
  - **UnbalancedQuotesQuery**: The query contains unbalanced quotes.
  - **BadProximityQuery**: The query contains an unrecognized proximity token.

  The error status is: `400` (BAD_REQUEST).

- Reviewing the related events on Sentry, I noticed that queries like this were raising **`BadProximityQuery`**:
`suitNature:"Securities"~1 AND caseName:"jpmorgan research equity long/short sel"~0.9`

  However, in this case, the problem is not the proximity tokens but the part **`long/short`**. This is a general syntax error due to the use of **`/`**, so it should raise a generic **`ElasticBadRequestError`** instead of **`BadProximityQuery`**.

  I have fixed **`check_for_proximity_tokens`** so strings like **`long/short`** are not detected as proximity token syntax errors.

Let me know what you think.

### CODE REVIEW

File: cl/search/exception.py [hunk 1]
```diff
@@ -47,7 +62,23 @@ class ElasticBadRequestError(APIException):
...
+class UnbalancedQuotesQueryAPIError(ElasticBadRequestError):
+    """APIException for handling unbalanced quotes in ES queries."""
+
+    default_detail = unbalanced_quotes_error_msg
+
+
+class BadProximityQueryAPIError(ElasticBadRequestError):
+    """APIException for handling bad proximity values in ES queries."""
+
+    default_detail = bad_proximity_query_error_msg
```
Comment (mlissner) [inline cl/search/exception.py:84]: Is there a way not to do these? They seem pretty clunky?

Comment (albertisfu) [reply cl/search/exception.py]: Yeah, the only thing that changes is the `detail` parameter. I've removed these exceptions and instead use `ElasticBadRequestError`, just passing their specific message.

Thanks for the suggestion.

**Review by mlissner — APPROVED** [review]
LGTM, thank you!

File: cl/search/exception.py (hunk 1 above)
Comment (mlissner) [reply cl/search/exception.py]: Thanks for the tweak! Much better now!

**Pull request merged**